src/components/App/AppIFrame.js

```diff
@@ -0,0 +1,154 @@
+  state = {
+    hideProgressBar: false,
+    loadProgress: 0,
+  }
+  componentWillMount() {
+    if (this.props.src) {
+      this.startProgress()
+    }
+  }
+  componentWillReceiveProps(nextProps) {
+    const { src: nextSrc } = nextProps
+    if (nextSrc !== this.props.src) {
+      this.resetProgress(() => {
+        // If we're setting src=undefined, don't start the progress bar as we're
+        // not actually loading a page
+        if (nextSrc) {
+          this.setProgressTimeout(this.startProgress, 100)
+        }
+      })
+    }
+  }
+  componentWillUnmount() {
+    this.clearProgressTimeout()
+  }
+  isHidden = () => {
+    const { hidden, src } = this.props
+    return !src || hidden
+  }
+  setProgressTimeout = (...args) => {
+    this.progressTimer = setTimeout(...args)
+  }
+  clearProgressTimeout = () => {
+    clearTimeout(this.progressTimer)
+  }
+  startProgress = () => {
+    this.setState({ loadProgress: LOADING_START }, () => {
+      this.setProgressTimeout(this.fudgeProgress, 500)
+    })
+  }
+  fudgeProgress = () => {
+    const { loadProgress } = this.state
+    if (loadProgress < LOADING_FUDGE_LIMIT) {
+      const delay = clamp(Math.random() * 1000, 350, 650)
+      // Move progress ahead by 1.5% to 7.5%
+      const moveProgress = clamp(Math.random() / 10, 0.02, 0.1)
+      const nextProgress = lerp(moveProgress, loadProgress, LOADING_END)
+      this.setState({ loadProgress: nextProgress }, () => {
+        this.setProgressTimeout(this.fudgeProgress, delay)
+      })
+    }
+  }
+  endProgress = () => {
+    this.clearProgressTimeout()
+    this.setState({ loadProgress: LOADING_END }, () => {
+      this.setProgressTimeout(() => {
+        this.setState({ hideProgressBar: true })
+      }, 1000)
+    })
+  }
+  resetProgress = (cb = noop) => {
+    this.clearProgressTimeout()
+    this.setState({ hideProgressBar: false, loadProgress: 0 }, cb)
+  }
+  handleOnLoad = (...args) => {
+    const { onLoad } = this.props
+    // As it turns out, setting src=undefined on an iframe also triggers the onLoad handler.
+    // We avoid doing anything in that case as the iframe hasn't really loaded a page.
+    if (this.props.src) {
+      this.endProgress()
+      if (typeof onLoad === 'function') {
+        onLoad(...args)
+      }
+    }
+  }
+  render() {
+    const { src, ...props } = this.props
+    const { hideProgressBar, loadProgress } = this.state
+    const show = !this.isHidden()
+    const progressBar = show && (
+      <AppLoadingProgressBar hide={hideProgressBar} percent={loadProgress} />
+    )
+
+    // Remove onLoad prop as we wrap it with our own
+    delete props.onLoad
+
+    return (
+      <React.Fragment>
+        {progressBar}
+        <StyledIFrame
+          frameBorder={0}
+          onLoad={this.handleOnLoad}
+          ref={iframe => {
+            this.iframe = iframe
+          }}
+          sandbox={SANDBOX}
+          style={{
+            display: show ? 'initial' : 'none',
```

Comment:
`initial` should be `block` here, otherwise it will revert back to `inline` (the default display for `iframe`) and a vertical scrollbar will appear.